sn_client/src/wallet.rs

```diff
@@ -159,25 +160,17 @@ impl WalletClient {
 
         let spent_ids: Vec<_> = transfer.tx.inputs.iter().map(|i| i.dbc_id()).collect();
 
-        let new_payment_proofs: PaymentProofsMap = audit_trail_info
-            .into_iter()
-            .map(|(addr, (audit_trail, path))| {
-                (
-                    addr,
-                    PaymentProof {
-                        spent_ids: spent_ids.clone(),
-                        audit_trail,
-                        path,
-                    },
-                )
-            })
-            .collect();
+        for addr in addrs_to_pay.into_iter() {
+            proofs.insert(
+                *addr,
+                PaymentTransactions {
+                    spent_ids: spent_ids.clone(),
```

Comment:
In case of paying for each chunk, then spend_id shall be different for each chunk_addr ?

I am more leaning towards to change the work flow within `fn pay_for_storage` to the following.
Which will testout the `pay for each chunk` directly with minium change.
```rust
pub async fn pay_for_storage(
    &mut self,
    content_addrs: impl Iterator<Item = &XorName>,
    verify_store: bool,
) -> Result<PaymentTransactionsMap> {
    // Let's filter the content addresses we hold payment proofs for, i.e. avoid
    // paying for those chunks we've already paid for with this wallet.
    let mut proofs = PaymentTransactionsMap::default();
    let addrs_to_pay: Vec<&XorName> = content_addrs
        .filter(|name| {
            if let Some(proof) = self.wallet.get_payment_proof(name) {
                proofs.insert(**name, proof.clone());
                false
            } else {
                true
            }
        })
        .collect();

    // If no addresses need to be paid for, we don't have to go further
    if addrs_to_pay.is_empty() {
        trace!("We already hold payment proofs for all the Chunks.");
        return Ok((proofs, None));
    }

    // Create payment proof for each chunk
    let mut new_payment_proofs = PaymentTransactionsMap::default();
    for addr in addres_to_pay.iter() {
        let storage_cost = self.get_store_cost_from_netowrk(addr).await?;
        info!("Storage cost for record {addr:?} is: {storage_cost}");

        // Create the payment spend for the chunk and send to network
        let transfer = self
            .wallet
            .local_send_storage_payment(storage_cost)
            .await?;
        trace!("Sending storage payment transfer to the network: {transfer:#?}");
        if let Err(error) = self.client.send(transfer.clone(), verify_store).await {
            warn!("The storage payment transfer was not successfully registered in the network: {error:?}. It will be retried later.");
            self.unconfirmed_txs.push(transfer);
            return Err(error);
        }

        if let Some(proof) = self.wallet.get_payment_proof(addr) {
            new_payment_proofs.insert(**addr, proof.clone());
        } else {
            warn!("Cannot get payment proof of {addr:?}, which is just got created");
        }
    }

    // cache the new set of payment proofs
    self.wallet.add_payment_proofs(new_payment_proofs.clone());

    // return the set of payment proofs, including new ones, and the ones we had in cache
    proofs.extend(new_payment_proofs);

    Ok(proofs)
}
```